tableprint/printer.py

```diff
@@ -18,12 +18,12 @@
 from six import string_types
 
 from .style import LineStyle, STYLES
-from .utils import ansi_len, format_line, parse_width
+from .utils import ansi_len, format_line, parse_width, calculate_widths
 
 __all__ = ('table', 'header', 'row', 'hrule', 'top', 'bottom', 'banner', 'dataframe', 'TableContext')
 
 STYLE = 'round'
-WIDTH = 11
```

Comment:
I had a look at this, but after working on it I felt that I was refactoring too much of "your" code and I think I respectfully withdraw. Specifically, I started moving the `mapdata` subfunction into `utils.py` since I think something like that is needed. I suspect you already have a better idea of how that refactoring is best handled.